pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -391,15 +391,22 @@ func (dswp *desiredStateOfWorldPopulator) createVolumeSpec(
 				return nil, "", err
 			}
 			// Error if a container has volumeMounts but the volumeMode of PVC isn't Filesystem
-			// or if a container has volumeDevices but the volumeMode of PVC isn't Block
-			if (mountsMap[podVolume.Name] && volumeMode != v1.PersistentVolumeFilesystem) ||
-				(devicesMap[podVolume.Name] && volumeMode != v1.PersistentVolumeBlock) {
+			if mountsMap[podVolume.Name] && volumeMode != v1.PersistentVolumeFilesystem {
 				return nil, "", fmt.Errorf(
-					"Invalid volumeMode. volumeMounts only supports 'Filesystem' mode and volumeDevices only supports 'Block' mode, pod: %q/%q, volume name: %v, volumeMode: %v",
+					"invalid combination of volumeMounts and volumeMode. volumeMounts expects volumeMode: 'Filesystem' but got '%v', pod: %q/%q, volume name: %v",
```

Comment:
A simpler error message could be something like "Volume %q has volumeMode %q, but is specified in volumeMounts for pod %q"

Reply:
ack.

Reply:
@msau42 
I'd like to have namespace for purpose of troubleshooting. Updated format is like this.
```
Volume "dswp-test-volume-name" has volumeMode "Filesystem", but is specified in volumeDevices for pod "dswp-test"/"dswp-test-pod"
Volume "dswp-test-volume-name" has volumeMode "Block", but is specified in volumeMounts for pod "dswp-test"/"dswp-test-pod"
```

Reply:
looks good, thanks!